packages/components/src/components/SearchBar.vue

```diff
@@ -185,78 +182,125 @@ export default {
+      }
+    },
+    decrementSelectedSuggestion() {
       if (
-        !getEventTarget(event.target, this.$refs.form) &&
-        this.$refs.input !== window.document.activeElement
+        this.selectedSuggestion !== false &&
+        this.selectedSuggestion !== this.suggestionsList.length - 1
       ) {
-        this.showSuggestions = false;
+        this.selectedSuggestion += 1;
+      } else {
+        this.selectedSuggestion = 0;
+      }
+    },
+    onWindowClick(event) {
+      if (!getEventTarget(event.target, this.$refs.form)) {
+        if (this.$refs.input !== window.document.activeElement) this.showSuggestions = false;
+
+        if (
+          !event.target.classList.contains('search-bar__suggestions-item') &&
+          !this.searchValue.endsWith(' ')
+        )
+          this.searchValue += ' ';
       }
     },
     onWindowKeyUp(event) {
       if (this.$refs.input !== window.document.activeElement) {
         return;
       }
+      if (this.keyDownTimeoutId !== undefined) {
+        clearTimeout(this.keyDownTimeoutId);
+        this.keyDownTimeoutId = undefined;
+      }
+      if (this.keyDownIntervalId !== undefined) {
+        clearInterval(this.keyDownIntervalId);
+        this.keyDownIntervalId = undefined;
+        return;
+      }
 
       switch (event.key) {
         case 'Escape':
           this.showSuggestions = false;
           this.$refs.input.blur();
+          if (!this.searchValue.endsWith(' ')) this.searchValue += ' ';
           break;
         case 'ArrowDown':
-          if (
-            this.selectedSuggestion !== false &&
-            this.selectedSuggestion !== this.suggestionsList.length - 1
-          ) {
-            this.selectedSuggestion += 1;
-          } else {
-            this.selectedSuggestion = 0;
-          }
+          this.decrementSelectedSuggestion();
           break;
         case 'ArrowUp':
-          if (this.selectedSuggestion !== false && this.selectedSuggestion !== 0) {
-            this.selectedSuggestion -= 1;
-          } else {
-            this.selectedSuggestion = this.suggestionsList.length - 1;
-          }
+          this.incrementSelectedSuggestion();
           break;
         default:
           this.selectedSuggestion = false;
           break;
       }
     },
+    onWindowKeyDown(event) {
+      if (
+        this.$refs.input !== window.document.activeElement ||
+        this.keyDownTimeoutId !== undefined
+      ) {
+        return;
+      }
+
+      this.keyDownTimeoutId = setTimeout(() => {
+        switch (event.key) {
+          case 'ArrowDown':
+            this.keyDownIntervalId = setInterval(this.decrementSelectedSuggestion.bind(this), 150);
+            break;
+          case 'ArrowUp':
+            this.keyDownIntervalId = setInterval(this.incrementSelectedSuggestion.bind(this), 150);
+            break;
+          default:
+            this.selectedSuggestion = false;
+            break;
+        }
+      }, 200);
+    },
```

Comment:
You need to call `event.preventDefault`, otherwise arrow keys move the text cursor in the field to beginning and end (at least on Firefox).

Reply:
Yup, good catch, thanks!